incubator/virtualcluster/pkg/syncer/controllers/pod/controller.go

```diff
@@ -222,7 +223,18 @@ func (c *controller) reconcilePodCreate(cluster, namespace, name string, pod *v1
 		return fmt.Errorf("failed to get secret: %v", err)
 	}
 
-	conversion.MutatePod(targetNamespace, pPod, vSecret, secret)
+	// list service in tenant ns and inject them into the pod
+	ca, err := innerCluster.GetCache()
```

Comment:
Can you just get services from Pod namespace?  The services outside of pod namespace should be irrelevant.

Reply:
kubelet will also add env from master service namespace.  see https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kubelet_pods.go#L537